### ISSUE TITLE
How to convert a list of variable object to respective list on with some modification 

### ISSUE THREAD

**Issue by sandeep020** [bazaarvoice/jolt#997]
I have to convert a list of location objects into elastic search should conditional object. The list of locations is variable at run time.
Input object type : 
 
{
  "values": [
    {
      "id": 1234
    },
    {
      "zip": "456765",
      "city": {
        "name": "Hyderabad"
      },
      "state": {
        "name": "Telangana"
      },
      "country": {
        "name": "India"
      }
    },
    {
      "city": {
        "name": "Banglore"
      },
      "state": {
        "name": "Karnataka"
      }
    }
  ]
}

and output object type : 

{
  "should": [
    {
      "id": 1234
    },
    {
      "must": [
        {
          "zip": 456765
        },
        {
          "city.name": "Hyderabad"
        },
        {
          "state.name": "Telangana"
        },
        {
          "country.name": "India"
        }
      ]
    },
    {
      "must": [
        {
          "city.name": "Banglore"
        },
        {
          "state.name": "Karnataka"
        }
      ]
    }
  ]
}
In short :  I have to use the index of object on left side to index of array on right side
